runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java

```diff
@@ -115,32 +117,51 @@ void refresh()
 	void triggerEvent(final DiscordGameEventType eventType)
 	{
 		final Optional<EventWithTime> foundEvent = events.stream().filter(e -> e.type == eventType).findFirst();
-		EventWithTime event;
+		final EventWithTime event;
 
 		if (foundEvent.isPresent())
 		{
 			event = foundEvent.get();
 		}
 		else
 		{
-			event = new EventWithTime(eventType, Instant.now());
-
+			event = new EventWithTime(eventType);
+			event.setStart(Instant.now());
 			events.add(event);
 		}
 
 		event.setUpdated(Instant.now());
 
-		if (event.getType().isShouldClear())
+		// An IN_GAME (or IN_MENU) event should remove all other areas from events
+		// but since events is fully cleared every time before IN_MENU is added,
+		// we will never need to run this on an IN_MENU event trigger
+		if (event.getType().isShouldClear() || event.getType() == DiscordGameEventType.IN_GAME)
```

Comment:
Imo this behaviour should be controlled by enum. Basically what you want here is flag to mark IN_GAME event to be "sticky" or something, e.g it should not be removed when shouldClear event triggers. Or add shouldBeCleared and shouldClear flags where shouldClear will clear other events that need to be cleared and shouldBeCleared will determine if the event will be cleared.

So in this case you would end up with IN_GAME event with shouldBeCleared: false, shouldClear: true